core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -52,10 +52,37 @@
 #include <HIP/Kokkos_HIP_Instance.hpp>
 #include <Kokkos_HIP.hpp>
 
+// TODO cannot use global variable on the device with ROCm 2.9
+//__device__ __constant__ unsigned long kokkos_impl_hip_constant_memory_buffer
+//    [Kokkos::Experimental::Impl::HIPTraits::ConstantMemoryUsage /
+//     sizeof(unsigned long)];
```

Comment:
Why should we prefer the global variable over the current implementation of `hip_parallel_launch_constant_memory()`?

Reply:
I don't think it's necessary to have a global memory for now but because it's only used to compute a guess block size but I think that it is necessary for other code that I haven't implemented yet.